base/bucket_gocb.go

```diff
@@ -1201,19 +1203,24 @@ func (bucket *CouchbaseBucketGoCB) UpdateXattr(k string, xattrKey string, exp ui
 		} else {
 			if cas == 0 {
 				// If the doc doesn't exist, set SubdocDocFlagMkDoc to allow us to write the xattr
-				mutateFlag = gocb.SubdocDocFlagMkDoc
+				if supportsTombstoneCreation {
+					mutateFlag = gocb.SubdocDocFlag(gocbcore.SubdocDocFlag(0x08)) | gocb.SubdocDocFlagAccessDeleted | gocb.SubdocDocFlagReplaceDoc
+				} else {
+					mutateFlag = gocb.SubdocDocFlagMkDoc
+				}
 			} else {
 				// Since the body _may_ not exist, we need to set SubdocDocFlagAccessDeleted
 				mutateFlag = gocb.SubdocDocFlagAccessDeleted
 			}
 		}
 
 		builder := bucket.Bucket.MutateInEx(k, mutateFlag, gocb.Cas(cas), exp).
-			UpsertEx(xattrKey, xv, gocb.SubdocFlagXattr).                                                // Update the xattr
+			UpsertEx(xattrKey, xv, gocb.SubdocFlagCreatePath|gocb.SubdocFlagXattr).                      // Update the xattr
 			UpsertEx(xattrCasProperty, "${Mutation.CAS}", gocb.SubdocFlagXattr|gocb.SubdocFlagUseMacros) // Stamp the cas on the xattr
+
 		if bucket.IsSupported(sgbucket.DataStoreFeatureCrc32cMacroExpansion) {
 			// Stamp the body hash on the xattr
-			if isDelete {
+			if isDelete && supportsTombstoneCreation {
```

Comment:
It does appear to default to 0x00 so we can go with that.

Reply:
Seems to default to `value_crc32c -> 0x00000000` which I guess equates to the same thing

Reply:
There is a test which checks for exactly 0x00 - Reviewing whether any actual production checks use this

Reply:
This is the SGW value which we store as a string and so although they equate to the same value it remains at the string value we set. I'm struggling to figure out the implications of it being `0x00000000` sometimes and `0x00` at other times. The only place I can see it being utilized is in `IsSGWrite` but if the doc is marked deleted we just set the cas to 0x00 anyway, rather than the body hash (as the body hash would be on {}). In the second delete op we also write `0x00` meaning there would be a difference there. 
I'm thinking we could just change the `0x00` `DeleteCrc32c` constant to `0x00000000` to avoid this issue. Would like your opinion on this.